### PR TITLE
Add Toggle component to Design System

### PR DESCRIPTION
**Pull request by trenaryja**
This PR adds a simple Toggle component to the Design System, which was migrated from the [Hotel Common Components](https://storybook.corp.priceline.com/?path=/story/hotel-common-components-atoms-togglebutton--default-behavior) 

Here is a screenshot of the Story for this component in the Design System Storybook:
image

### CODE REVIEW

File: packages/core/src/Toggle/Toggle.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import React from 'react'
+import { Toggle } from '.'
+import { render, fireEvent } from '../__test__/testing-library'
+
+describe('Toggle', () => {
+  test('renders correctly when off', () => {
+    const { getByRole } = render(<Toggle label='Total price toggle' />)
+    const container = getByRole('switch').parentNode
+    expect(container).toHaveStyleRule('background-color', '#4f6f8f')
```
Comment (craigpalermo) [inline packages/core/src/Toggle/Toggle.spec.tsx:9]: Can you pull the values used for the assertions from the theme?

File: packages/core/src/Toggle/Toggle.tsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  isOn,
+  label,
+  onToggle,
+  disabled,
+  height,
+  icon,
+}) => {
+  const width = height * 1.875
+  const borderRadius = `${height / 2}`
+  const outlineAbsoluteBorderRadius = `${height + 4}`
```
Comment (craigpalermo) [inline packages/core/src/Toggle/Toggle.tsx:72]: I wonder if there's anything we need to change here in light of the recent `borderRadii` updates

File: packages/core/src/Toggle/Toggle.spec.tsx (hunk 1 above)
Comment (trenaryja) [reply packages/core/src/Toggle/Toggle.spec.tsx]: Great idea, updated to import colors from theme!

File: packages/core/src/Toggle/Toggle.tsx (hunk 2 above)
Comment (trenaryja) [reply packages/core/src/Toggle/Toggle.tsx]: Updated to add use of borderRadius prop! @craigpalermo could you possibly take another look to see if this is closer to what you may have been envisioning?

File: packages/core/src/Toggle/Toggle.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  test('renders correctly when off', () => {
+    const { getByRole } = render(<Toggle label='Total price toggle' />)
+    const container = getByRole('switch').parentNode
+    expect(container).toHaveStyleRule('background-color', colors.gray)
+  })
+
+  test('renders correctly when on', () => {
+    const { getByRole } = render(<Toggle isOn label='Total price toggle' />)
+    const container = getByRole('switch').parentNode
+    expect(container).toHaveStyleRule('background-color', colors.blue)
```
Comment (craigpalermo) [inline packages/core/src/Toggle/Toggle.spec.tsx:16]: I think these should use palette colors; the named colors are deprecated

File: packages/core/src/Toggle/Toggle.tsx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  top: 0;
+  left: 0;
+  margin: 0;
+  cursor: ${(props) => (props.disabled ? 'default' : 'pointer')};
+`
+
+const OutlineAbsolute = styled(Absolute)`
+  height: calc(100% + 8px);
+  opacity: 0;
+  background-color: ${(props) => alphaColor(props.isOn ? colors.blue : colors.gray)};
```
Comment (sdalonzo) [inline packages/core/src/Toggle/Toggle.tsx:23]: these colors, too

File: packages/core/src/Toggle/Toggle.tsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+  transition: left 0.3s;
+`
+
+const WrapperBox = styled(Box)`
+  position: relative;
+  height: ${(props) => props.height}px;
+  transition: background-color 0.3s;
+  min-width: ${(props) => props.width}px;
+
+  ${AbsoluteInput}:focus + ${OutlineAbsolute} {
```
Comment (sdalonzo) [inline packages/core/src/Toggle/Toggle.tsx:41]: styled-components super powers!

File: packages/core/src/Toggle/Toggle.spec.tsx (hunk 3 above)
Comment (trenaryja) [reply packages/core/src/Toggle/Toggle.spec.tsx]: Updated to use palette colors 👍

File: packages/core/src/Toggle/Toggle.tsx (hunk 4 above)
Comment (trenaryja) [reply packages/core/src/Toggle/Toggle.tsx]: Updated to use palette colors 👍

**Review by sdalonzo — APPROVED** [review]

**Review by craigpalermo — APPROVED** [review]

**Pull request merged**